mitodl/open-discussions
# Opensearch Client (PR 3 of 3)

#### Pre-Flight checklist


#### What are the relevant tickets?
updates https://github.com/mitodl/open-discussions/issues/3494

#### What's this PR do?
This PR swaps the Elasticsearch client out for OpenSearch. This will run the OpenSearch client against the Elasticsearch service.  The next and final PR in the series will update the service.

#### How should this be manually tested?
Pull the branch locally and run docker build and docker up to change your client to OpenSearch. The Docker container (service) will not change.

I would then:

run tests
check that you can use the search API manually using postman or fetch or some tool of your choice - specifically as OCW might make requests
make sure that pages within Open that are using the search API function (new open demo, infinite, learn, open discussions)

#### Where should the reviewer start?
Pull the branch and play with it, honestly. I can run all the tests in the world and I can try to break it, but ultimately it's best if we have many people poke it from all sides.

#### Any background context you want to provide?
Testing in a similar manner to the ES7 PR would probably be a good start!
